text/0000-grammar-is-canonical.md

```diff
@@ -0,0 +1,97 @@
+
+This RFC proposes following approach to making [src/grammar] the canonical Rust language grammar:
+
+1. Fix the already known discrepancies between implementation and [src/grammar];
+2. Make [src/grammar] a [semi-canonical grammar];
+3. After a period of time transition [src/grammar] to a [fully-canonical grammar].
+
+## Semi-canonical grammar
+[semi-canonical grammar]: #semi-canonical-grammar
+
+Once all known discrepancies between the [src/grammar] and rustc parser implementation are
+resolved, [src/grammar] enters the state of being semi-canonical grammar of the Rust language.
+
+Semi-canonical means that all new development involving syntax changes are made and discussed in
+terms of changes to the [src/grammar] and [src/grammar] is in general regarded to as the canonical
+grammar except when new discrepancies are discovered. These discrepancies must be swiftly resolved,
+but resolution will depend on what kind of discrepancy it is:
+
+1. For syntax changes/additions introduced after [src/grammar] gained the semi-canonical state, the
+   [src/grammar] is canonical;
+2. For syntax that was present before [src/grammar] gained the semi-canonical state, in most cases
+   the implementation is canonical.
+
+This process is sure to become ambiguous over time as syntax is increasingly adjusted (it is harder
+to “blame” syntax changes compared to syntax additions), therefore the resolution process of
+discrepancies will also depend more on a decision from the Rust team.
+
+## Fully-canonical grammar
+[fully-canonical grammar]: #fully-canonical-grammar
+
+After some time passes, [src/grammar] will transition to the state of fully canonical grammar.
+After [src/grammar] transitions into this state, for any discovered discrepancies the
+rustc parser implementation must be adjusted to match the [src/grammar], unless decided otherwise
+by the RFC process.
+
+## RFC process changes for syntactic changes and additions
+
+Once the [src/grammar] enters semi-canonical state, all RFCs must describe syntax additions and
+changes in terms of the formal [src/grammar]. Discussion about these changes are also expected (but
+not necessarily will) to become more formal and easier to follow.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This RFC introduces a period of ambiguity during which neither implementation nor [src/grammar] are
+truly canonical representation of the Rust language. This will be less of an issue over time as
+discrepancies are resolved, but its an issue nevertheless.
+
+# Alternatives
+[alternatives]: #alternatives
+
+One alternative would be to immediately make [src/grammar] a fully-canonical grammar of the Rust
+language at some arbitrary point in the future.
+
+Another alternative is to simply forget idea of having a formal grammar be the canonical grammar of
+the Rust language.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+How much time should pass between [src/grammar] becoming semi-canonical and fully-canonical?
```

Comment:
I feel there does not have to be a defined time period per se. We'll work it out.